store/mongo/datastore_mongo.go

```diff
@@ -1306,13 +1305,36 @@ func (db *DataStoreMongo) UpdateDeviceDeploymentLogAvailability(ctx context.Cont
 	return nil
 }
 
+// SaveDeviceDeploymentRequest saves device deployment request
+// with the device deployment object
+func (db *DataStoreMongo) SaveDeviceDeploymentRequest(
+	ctx context.Context,
+	ID string,
+	request *model.DeploymentNextRequest,
+) error {
+
+	database := db.client.Database(mstore.DbFromContext(ctx, DatabaseName))
+	collDevs := database.Collection(CollectionDevices)
+
+	res, err := collDevs.UpdateOne(
+		ctx,
+		bson.D{{Key: StorageKeyId, Value: ID}},
+		bson.D{{Key: "$set", Value: bson.M{StorageKeyDeviceDeploymentRequest: request}}},
```

Comment:
Could there be instances where the device deployment document does not (yet) exist?

Reply:
to be able to save the request you have to have the device deployment ID, so I think the object should be always there